syncd/FlexCounter.cpp

```diff
@@ -1367,6 +1417,54 @@ void FlexCounter::collectPriorityGroupAttrs(
     }
 }
 
+void FlexCounter::collectMACsecSAAttrs(
+        _In_ swss::Table &countersTable)
+{
+    SWSS_LOG_ENTER();
+
+    // Collect attrs for every registered MACsec SA
+    for (const auto &kv: m_macsecSAAttrIdsMap)
+    {
+        const auto &macsecSAVid = kv.first;
+        const auto &macsecSARid = kv.second->m_macsecSAId;
+        const auto &macsecSAAttrIds = kv.second->m_macsecSAAttrIds;
+
+        std::vector<sai_attribute_t> macsecSAAttr(macsecSAAttrIds.size());
+
+        for (size_t i = 0; i < macsecSAAttrIds.size(); i++)
+        {
+            macsecSAAttr[i].id = macsecSAAttrIds[i];
+        }
+
+        // Get MACsec SA attr
+        sai_status_t status = m_vendorSai->get(
+                SAI_OBJECT_TYPE_MACSEC_SA,
+                macsecSARid,
+                static_cast<uint32_t>(macsecSAAttr.size()),
+                macsecSAAttr.data());
+
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_WARN("Failed to get attr of MACsec SA 0x%" PRIx64 ": %d", macsecSAVid, status);
+            continue;
+        }
+
+        // Push all counter values to a single vector
+        std::vector<swss::FieldValueTuple> values;
+
+        for (size_t i = 0; i != macsecSAAttrIds.size(); i++)
+        {
+            const std::string &attrName = sai_serialize_enum(macsecSAAttrIds[i], &sai_metadata_enum_sai_macsec_sa_attr_t);
+            auto meta = sai_metadata_get_attr_metadata(SAI_OBJECT_TYPE_MACSEC_SA, macsecSAAttr[i].id);
+            values.emplace_back(attrName, sai_serialize_attr_value(*meta, macsecSAAttr[i]));
+        }
+        // Write counters to DB
+        std::string macsecSAVidStr = sai_serialize_object_id(macsecSAVid);
```

Comment:
just a note, every file uses concept of prefixing "strXXX" instead of "xxxStr", this file is exception since i didn't created this, but in other files please follow strXXX naming